lib/replicator.js

```diff
@@ -1233,42 +1235,10 @@ class Peer {
     return false
   }
 
-  _hasTreeParent (index) {
+  _canRequest (index) {
```

Comment:
What do you think of leaving the assertion `if (!(index >= 0)) throw ASSERTION('bad index to _hasTreeParent: ' + index)`, in case a negative index was indeed the cause of the issue? Because if the index is ever negative, it indicates a bug in some other part of the code.

Reply:
Done